### ISSUE TITLE
License Issue when Offline

### ISSUE THREAD

**Issue by technobird22** [wangyw15/CeVIO_Crack#2]
When online, it shows error upon startup `Since an problem occurred in the Activation Server, it starts offline.`, but all features work normally :)

When offline, after the error is shown, it asks for license keys, then closes automatically, saying it does not detect any installed voices.

Could it be made working even when offline?

CeVIO AI version: 9.1.9.0
Activator version: v1.5

Thanks for your work on this!

**Comment by wangyw15** [comment]
"When offline" means that the system is disconnected from network?
If this condition, I haven't considered of it, need some time to work.

**Comment by wangyw15** [comment]
I tried block the program from accessing network. It warn the following message:

```
Unable to connect to the Activation Server:

- There is a possibility that internet communication is blocked by security software.

- There is a possibility that Port 443 is occupied by Virtualization Software.
```

Is that the message you encountered?

**Comment by technobird22** [comment]
Hmm, I haven't encountered those messages before. Yes, by offline I mean the machine is disconnected from the network.

What I've found is that it still shows the `Since an problem occurred in the Activation Server, it starts offline.` error (which is also shown when it is connected to the network and works), but after displaying the error, it requests me to activate the voices (fig 1), then says that there are no voices installed and exits (fig 2).

Figure 1:
![image](https://github.com/user-attachments/assets/75529ff9-9ea1-4fe6-a694-70b249a330fe)

Figure 2:
![image](https://github.com/user-attachments/assets/c4919c27-2a89-4c0a-a659-6e0324c8b7c4)

**Comment by wangyw15** [comment]
I've reproduced the problem. It's kind of strange, because there's code to start offline, but some checks block the offline start.

Need more time to inspect, but it's too difficult.

**Comment by wangyw15** [comment]
I think there's never a real offline startup, because it still needs network connection for genuine license.

I tried to modify the executable itself, but it refused to start after modification.

**Comment by technobird22** [comment]
Yes, I agree with you, although in older versions of the CeVIO editor, it was possible to still run and export offline using the unofficial CeVIO SFE tool, if I remember correctly.

Since you've managed to get it bypassed and working when it is online, I wonder if there's a way to bypass the network check so it works offline too.

I didn't think there would be too big of a difference between it being online and offline, so wonder if it's possible to bypass offline start too.

I agree that it may be hard to achieve, but if it is possible, that would be awesome.

Thanks for your hard work!

**Comment by wangyw15** [comment]
Yeah, both activation process is identical. The editor allows up to 1 year for offline usage. but if it cannot access network, it sees today as 12/31/9999.

I've thought out a temporary solution: for offline use, activate to 12/31/9999, for online use, activate to 1 year later. Strange but it might work.

**Comment by wangyw15** [comment]
I tried to inspect how SFE works, but the patch itself is obfuscated :(

**Comment by wangyw15** [comment]
I've updated and released a new version, you can try as the new release described.

[v1.6](https://github.com/wangyw15/CeVIOActivator/releases/tag/v1.6)

**Comment by wangyw15** [comment]
Due to the bypass method, it may raise new problems. Feel free to open new issue :)

**Comment by technobird22** [comment]
Awesome, thanks! I'll try it out when I have the chance :)

**Comment by wangyw15** [comment]
[v2.0](https://github.com/wangyw15/CeVIOActivator/releases/tag/v2.0) is released, it should open without any warning.

Feel free to open new issues.
